### ISSUE TITLE
Install PowerShell and ENABLE_PSREMOTING without service restart

### ISSUE THREAD

**Issue by brwilkinson** [PowerShell/PowerShell#12225]
Is there an argument where it's possible to Enable_psremoting, however flag for a reboot instead of restarting winrm?

arguments are specified here:
https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell-core-on-windows?view=powershell-7

E.g.
'ADD_EXPLORER_CONTEXT_MENU_OPENPOWERSHELL=1 ENABLE_PSREMOTING=1 REGISTER_MANIFEST=1'

**Comment by iSazonov** [comment]
I guess it is possible if we would use Set-WSManQuickConfig cmdlet (See https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/enable-psremoting?view=powershell-7) instead of Enable-PSRemoting.

/cc @PaulHigin

**Comment by heaths** [comment]
Admins and users alike generally hate reboots, though they are a fact of the way Windows puts files in use. Why would you want to force a reboot (even if optional) rather than restart WinRM? Rebooting the machine will also down WinRM for a time - longer even. What's the use-case for supporting this?

**Comment by brwilkinson** [comment]
@heaths 

We deploy using DSC (Desired State Configuration) . . So the upgrade of PowerShell breaks the DSC deployment since it restarts WINRM.

In DSC the philosophy is to set the reboot flag and then handle it later in the XPendingReboot resource. Similar how packages have /norestart. This would also work well for and ENABLE_PSREMOTING=1 RESTART_WIN RM=0

**Comment by iSazonov** [comment]
@SteveL-MSFT @TravisEz13 Makesense to support the DSC install scenario?

**Comment by SteveL-MSFT** [comment]
It seems ok to add a `-NoWinRmRestart` type switch to `Enable-PSRemoting` cmdlet to skip the restart of WinRM and expose that as a MSI arg.

**Comment by iSazonov** [comment]
@brwilkinson Do you want to pull PR?

**Comment by brwilkinson** [comment]
I have not contributed to date, however I pulled down the repo and built the project yesterday.

I would likely need a mentor to get me through the process end to end... however I can make a start at looking into the code for the PSRemoting.

Feedback on the build process...(only a small thing).
Not sure why I had to create the emtpy nuget directories for the Modules that are built within the project . .. C:\Users\<user>\.nuget\packages.. e.g. psreadline,theadjob,psdesiredstateconfiguration,microsoft.powershell.archive,packagemanagement,powershellget.    <-- not sure if this is a bug on the build to create these dirs or not ?

**Comment by iSazonov** [comment]
> I would likely need a mentor to get me through the process end to end... however I can make a start at looking into the code for the PSRemoting.

Yes, maintainers will help you. Look merged PRs as examples. Also we need test(s) but I don't see tests for Enable-PSRemoting. Maybe a manual check is enough for us. /cc @PaulHigin 

Currently you need find sources for Enable-PSRemoting and add new parameter NoWinRmRestart.

**Comment by iSazonov** [comment]
> Feedback on the build process...(only a small thing).

I never see this. I'd expect that `dotnet restore` creates the folders. `Start-PSBuild -Restore` runs  `dotnet restore` for all projects.
My experience - I always install latest Visual Studio and needed .Net Core from msi. This ensures stable build for .Net Core previews on VS Code.

**Comment by timoline** [comment]
Any progress on this?

same proble here, the upgrade of PowerShell breaks the DSC deployment

**Comment by jberezanski** [comment]
I just encountered a situation where installing PowerShell 7 with ENABLE_PSREMOTING using DSC caused the WinRM service to hang in the "StopPending" state. While in this state, the service rejected all connection attempts with HTTP error code 500 (Internal Server Error). After 40 minutes I finally decided to log on via RDP and manually kill the relevant svchost process - this allowed the DSC configuration process to proceed without further problems. This happened on four different servers, running Windows Server 2019 and 2022 (both Datacenter Core).

I'm very much interested in the proposed "no WinRM restart" flag; I might take a stab at implementing it. I took a cursory look at the implementation of `Enable-PSRemoting` and I can see one complication: the cmdlet performs several actions, some of which restart the service (Enable-PSSessionConfiguration, Register-PSSessionConfiguration). It's not clear to me yet if the subsequent actions depend on the service having been restarted. Perhaps some restructuring of the code needs to be done.
